Error/ErrorHandler.php

```diff
@@ -26,13 +26,17 @@ class ErrorHandler
     /** @var string */
     private $internalErrorMessage;
 
-    public function __construct($internalErrorMessage = null, LoggerInterface $logger = null)
+    /** @var array */
+    private $exceptionMap;
+
+    public function __construct($internalErrorMessage = null, LoggerInterface $logger = null, array $exceptionMap = [])
     {
         $this->logger = (null === $logger) ? new NullLogger() : $logger;
         if (empty($internalErrorMessage)) {
             $internalErrorMessage = self::DEFAULT_ERROR_MESSAGE;
         }
         $this->internalErrorMessage = $internalErrorMessage;
+        $this->exceptionMap = $exceptionMap;
```

Comment:
`exceptionMap` could be transformed to improve the lookup. Currently `convertException` must loop over every entries to find the error class. If we flatten it we could greatly improve the lookup.

```php
// e.g.
[
  'Symfony\Component\Routing\Exception\ResourceNotFoundException' => 'warning',
  'InvalidArgumentException' => 'error',
]